script/eslint-plugin-va/rules/prefer-web-component-library.js

```diff
@@ -0,0 +1,60 @@
+const MESSAGE =
+  '<{{ reactComponent }}> can be replaced by <{{ webComponent }}>';
+
+const getPropNode = (node, propName) =>
+  node.openingElement.attributes.find(n => n.name.name === propName);
+
+const telephoneTransformer = (context, node) => {
+  const componentName = node.openingElement.name;
+  const patternNode = getPropNode(node, 'pattern');
+  const notClickableNode = getPropNode(node, 'notClickable');
+  const international =
+    patternNode?.value.expression.property.name === 'OUTSIDE_US';
+
+  context.report({
+    node,
+    message: MESSAGE,
+    data: {
+      reactComponent: componentName.name,
+      webComponent: 'va-telephone',
+    },
+    fix: fixer => {
+      // Replace the node name
+      // and remove the `pattern` prop if it's there
+      return [
+        fixer.replaceText(componentName, 'va-telephone'),
+        international
+          ? fixer.insertTextBefore(patternNode, 'international')
+          : null,
+        patternNode ? fixer.remove(patternNode) : null,
+        notClickableNode
+          ? fixer.replaceText(notClickableNode, 'not-clickable')
+          : null,
+      ].filter(i => !!i);
```

Comment:
This is the equivalent to what the transformer function would do if we implemented this as part of the original migration script.